File: scala/src/test/scala/org/agilegrenoble/objectcalisthenics/GildedRoseTest.scala
package org.agilegrenoble.objectcalisthenics

import java.util.Random
import GildedRoseTest._
import org.scalatest.matchers.ShouldMatchers
import org.scalatest.FlatSpec
import org.scalatest.BeforeAndAfter
import org.scalatest.matchers.Matcher
import org.scalatest.matchers.MatchResult
import scala.math.Ordered

object GildedRoseTest {

}

class GildedRoseTest extends FlatSpec with ShouldMatchers with BeforeAndAfter {

  private var gildedRose: GildedRose = _

  before {
    gildedRose = new GildedRose()
  }

  "GildedRose" should "look a certain way after day one" in {
    gildedRose.updateQuality()
    val items = gildedRose.items
    //items[0] should equal(...)
  }

}

File: javascript/test/GildedRose.test.js
var chai = require("chai"),
    assert = chai.assert,
    GildedRose = require('../src/GildedRose.js'),
    Item = require('../src/Item.js'),
    MersenneTwister = require('./lib/mersenne-twister.js');

chai.should();

suite('GildedRose', function () {

    var gildedRose,
        items

    setup(function () {
        gildedRose = new GildedRose();
        items = gildedRose.makeItems();
    });

    test('after one day', function () {
        gildedRose.updateQuality(items)
        // assert.equal(items[0], ...)
    });

});
